CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1505,12 +1505,7 @@ protected function postProcessMembership(
 
         if (!empty($membershipContribution)) {
           // Next line is probably redundant. Checks prevent it happening twice.
-          $membershipPaymentParams = [
-            'membership_id' => $membership->id,
-            'membership_type_id' => $membership->membership_type_id,
-            'contribution_id' => $membershipContribution->id,
-          ];
-          civicrm_api3('MembershipPayment', 'create', $membershipPaymentParams);
+          CRM_Member_BAO_MembershipPayment::legacyMembershipPaymentCreate($membership->id, $membershipContribution->id);
```

Comment:
I wonder if we could check if this does anything & maybe remove it. I'm pretty sure creating the line item does this

Reply:
Yeah, the extracted function is only called in one place. But my thinking is that it was easier to extract everything related to MembershipPayment first and then it should be easier to justify just removing stuff. Eg. if `getLatestContributionIDFromLineitemAndFallbackToMembershipPayment()` function is used everywhere we can probably justify just removing this call to MembershipPayment.create on the basis that we don't need the MembershipPayment record to be created at all as we can verify that a Membership `LineItem` *is* created and we know that the LineItem will be checked first.